### PR TITLE
chore: remove support for node 8-12, add 18

### PR DESCRIPTION
**Pull request by Rauno56**
## Which problem is this PR solving?

It's a complimentary PR for https://github.com/open-telemetry/opentelemetry-js/pull/3048

## Short description of the changes

Required changes for node@18:

- 1e6bddf481f6e9201a0f77edd4596da987e2c41e: turned off tests for alibaba detector **in node@18**.
- [fceb7b7](https://github.com/open-telemetry/opentelemetry-js-contrib/pull/1065/commits/fceb7b708235bd677de04f064962269c09978a9a): turned off tests for fastify instrumentation **in node@18**
- https://github.com/open-telemetry/opentelemetry-js-contrib/pull/1065/commits/0067aecf2cf725387c4dfc0f692cef274ffe21be: error message used in tests changed
- https://github.com/open-telemetry/opentelemetry-js-contrib/pull/1065/commits/ef29bcfdde997232038fe0817e409baf4bd43113: Changed how root context equality is checked. Check the commit message.

## Checklist

- [ ] Ran `npm run test-all-versions` for the edited package(s) on the latest commit if applicable.

### CODE REVIEW

File: detectors/node/opentelemetry-resource-detector-alibaba-cloud/src/detectors/AlibabaCloudEcsDetector.ts [hunk 1]
```diff
@@ -114,7 +114,7 @@ class AlibabaCloudEcsDetector implements Detector {
           typeof statusCode !== 'number' ||
           !(statusCode >= 200 && statusCode < 300)
         ) {
-          res.destroy();
+          res.socket.destroy();
```
Comment (blumamir) [inline detectors/node/opentelemetry-resource-detector-alibaba-cloud/src/detectors/AlibabaCloudEcsDetector.ts:117]: why this change?

**Review by blumamir — APPROVED** [review]

File: detectors/node/opentelemetry-resource-detector-alibaba-cloud/src/detectors/AlibabaCloudEcsDetector.ts (hunk 1 above)
Comment (Rauno56) [reply detectors/node/opentelemetry-resource-detector-alibaba-cloud/src/detectors/AlibabaCloudEcsDetector.ts]: I saw a cryptic error message and immediately assumed that's because of a changed node API, since the only variable seemed to be node's version.

It actually might be because of `nock` as well, I don't know. I'll revert the change and turn off tests for node@18 for the detector for now. Thanks for questioning this!

Error message:

```
     Uncaught TypeError: The "stream" argument must be an instance of Stream. Received an instance of Socket
      at new NodeError (node:internal/errors:388:5)
      at eos (node:internal/streams/end-of-stream:65:11)
      at IncomingMessage._destroy (node:_http_incoming:234:21)
      at _destroy (node:internal/streams/destroy:109:10)
      at IncomingMessage.destroy (node:internal/streams/destroy:71:5)
      at OverriddenClientRequest.<anonymous> (src/detectors/AlibabaCloudEcsDetector.ts:122:15)
      at Object.onceWrapper (node:events:652:26)
      at OverriddenClientRequest.emit (node:events:537:28)
      at Timeout.respond [as _onTimeout] (node_modules/nock/lib/playback_interceptor.js:307:11)
      at listOnTimeout (node:internal/timers:564:17)
      at processTimers (node:internal/timers:507:7)
```

cc: @legendecas

**Review by dyladan — APPROVED** [review]
